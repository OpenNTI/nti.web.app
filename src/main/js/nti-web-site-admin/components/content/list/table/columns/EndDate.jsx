import './EndDate.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';
import {DateTime} from '@nti/web-commons';

const t = scoped('nti-web-site-admin.components.content.list.table.columns.EndDate', {
	title: 'End Date'
});

export default class EndDate extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired
	}

	static cssClassName = 'enddate-col';

	static Name = () => t('title')

	static SortKey = 'endDate';

	render () {
		const {item} = this.props;
		return (
			<div className="cell">
				<DateTime className="value" date={item.getEndDate()} format={DateTime.MONTH_ABBR_DAY_YEAR} />
			</div>
		);
	}
}
