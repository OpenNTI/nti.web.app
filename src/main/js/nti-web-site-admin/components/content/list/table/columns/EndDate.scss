@import '~@nti/style-common';

.content-table-container {
	.enddate-col {
		color: var(--secondary-grey);
		font-size: 14px;
		line-height: 60px;
		width: 7rem;
	}
}
