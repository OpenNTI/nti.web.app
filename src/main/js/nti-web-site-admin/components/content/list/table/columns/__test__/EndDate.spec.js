/* eslint-env jest */
import { act, create } from 'react-test-renderer';

import EndDate from '../EndDate';

describe('Site admin content table end date column test', () => {
	const verifyColumn = date => {
		let cmp;
		act(() => {
			cmp = create(
				<EndDate
					item={{
						getEndDate: () => {
							return date;
						},
					}}
				/>
			);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
		cmp.unmount();
	};

	test('Test date is many days ago', () => {
		// label should show date formatted as "Oct 31, 2017"
		verifyColumn(new Date('10/31/2017'));
	});
});
