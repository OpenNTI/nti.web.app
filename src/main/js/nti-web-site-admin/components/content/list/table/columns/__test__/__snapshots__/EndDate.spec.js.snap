// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin content table end date column test Test date is many days ago 1`] = `
<div
  className="cell"
>
  <time
    className="ntiText value"
    dateTime="2017-10-31T00:00:00+00:00"
  >
    Oct 31, 2017
  </time>
</div>
`;
